h2/src/main/org/h2/server/pg/PgServer.java

```diff
@@ -339,8 +340,12 @@ public static String formatType(int pgType) {
         case PG_TYPE_FLOAT8:
             valueType = Value.DOUBLE;
             break;
+        case PG_TYPE_INT2_ARRAY:
+            return "_int2";
+        case PG_TYPE_INT4_ARRAY:
+            return "_int4";
         case PG_TYPE_VARCHAR_ARRAY:
-            return "character varying[]";
+            return "_varchar";
```

Comment:
Where these strings came from? The real PostgreSQL returns `smallint[]`, `integer[]`, and `character varying[]` from its `format_type()` function.

Reply:
I thought they came from `pg_type.typname` :-P